src/tests/ftest/util/host_utils.py

```diff
@@ -0,0 +1,589 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from logging import getLogger
+from getpass import getuser
+import os
+from re import findall
+from socket import gethostname
+
+from avocado import fail_on
+from avocado.utils import process
+from ClusterShell.NodeSet import NodeSet, NodeSetParseError
+
+from agent_utils import AgentManager, AgentCommand, AgentYamlParameters
+from agent_utils import stop_agent_processes
+from command_utils import ObjectWithParameters, BasicParameter, YamlParameters
+from command_utils import CommandFailure, EnvironmentVariables
+from general_utils import check_file_exists
+from logger_utils import TestLogger
+from server_utils import ServerManager, ServerCommand, ServerYamlParameters
+from server_utils import stop_server_processes
+
+
+def get_partition_hosts(partition):
+    """Get a list of hosts in the specified slurm partition.
+
+    Args:
+        partition (str): name of the partition
+
+    Returns:
+        list: list of hosts in the specified partition
+
+    """
+    log = getLogger()
+    hosts = []
+    if partition is not None:
+        # Get the partition name information
+        cmd = "scontrol show partition {}".format(partition)
+        try:
+            result = process.run(cmd, shell=True, timeout=10)
+        except process.CmdError as error:
+            log.warning(
+                "Unable to obtain hosts from the {} slurm "
+                "partition: {}".format(partition, error))
+            result = None
+
+        if result:
+            # Get the list of hosts from the partition information
+            output = result.stdout
+            try:
+                hosts = list(NodeSet(findall(r"\s+Nodes=(.*)", output)[0]))
+            except (NodeSetParseError, IndexError):
+                log.warning(
+                    "Unable to obtain hosts from the {} slurm partition "
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments